spiceConcurrentCall_test.py

```diff
@@ -0,0 +1,109 @@
+# %%
+import spiceypy as spice # this imports the version 4.0.0
+import os
+
+# %%
+import functools
+import threading
+
+from spiceypy.spiceypy import spkezr
+
+_spicelock = threading.RLock()
+
+enablethreadinglock = False
+
+def spicelock_for_multithread(f):
+    """
+    Decorator for spiceypy to avoid concurrent calls to cspice lib.
+
+    :return:
+    """
+    @functools.wraps(f)
+    def lock(*args, **kwargs):
+        if enablethreadinglock:
+            with _spicelock:
+                try:
+                    res = f(*args, **kwargs)
+                    return res
+                except BaseException:
+                    raise
+        else:
+            try:
+                res = f(*args, **kwargs)
+                return res
+            except BaseException:
+                raise
+
+    return lock
+
+# %%
+from multiprocessing.dummy import Pool as ThreadPool
+
+
+
+# %%
+b1900 = spice.b1900
+tb1900 = spicelock_for_multithread(b1900)
+
+# %%
+print('Test thread pool without locking for b1900')
+spice.enablethreadinglock = False
+with ThreadPool() as pool:
+    test_b1900 = pool.map(lambda x: tb1900(), range(10000))
+
+print('Test thread pool with locking for b1900')
+spice.enablethreadinglock = True
+with ThreadPool() as pool:
+    test_tb1900 = pool.map(lambda x: tb1900(), range(10000))
+
+
+
+# %%
+pathKernels = 'path/to/kernels'
+
+defKernels = [
+    'naif0012.tls',                     # leaps seconds
+    'gm_de431.tpc',                     # planetary constants
+    'de430.bsp',                        # planetary position and velocity ephemeris
+    'earth_fixed.tf',                   # ECEF reference system definition
+    'earth_070425_370426_predict.bpc',  # ECEF ephemeris 
+    'pck00010.tpc'                      # Other constants
+]
+
+for k in defKernels:
+    spice.furnsh(os.path.join(pathKernels,k))
+
+# %%
+spkezr = spice.spkezr
+tspkezr = spicelock_for_multithread(spkezr)
+# %%
+# testing the spicelock_for_multithread decorator
+print('Test thread pool without locking for spkezr')
+spice.enablethreadinglock = False
+try:
+    with ThreadPool() as pool:
+        test_spkezr = pool.map(lambda x: tspkezr('EARTH',x,'J2000','NONE','SUN'),test_b1900)
+except BaseException as e:
+    print(e)
+
+
+print('Test thread pool with locking for spkezr')
+spice.enablethreadinglock = True
```

Comment:
I was actually envisioning a new set of contextmanagers for the enabling/disabling of the use of your new decorator like I have implemented for the "found" flag checking, see: https://github.com/AndrewAnnex/SpiceyPy/blob/main/spiceypy/spiceypy.py#L150. I also think adding trivial getter/setter/checker functions like I have for the other setting is nicer to use than directly interacting with a parameter like done here.

Reply:
I implemented the entire suite of functions to handle the behavior as "found" with the context manager to avoid direct modification of the parameter